sdks/common/runner-api/src/main/proto/beam_runner_api.proto

```diff
@@ -100,45 +96,57 @@ message Pipeline {
   // this pipeline.
   Components components = 1;
 
-  // (Required) The graph node that is the root of the graph.
-  string root_graph_node = 2;
+  // (Required) The id of the PTransform that is the root of the pipeline
+  string root_transform_id = 2;
 
-  // (Required) Static display data for the pipeline.
+  // (Optional) Static display data for the pipeline. If there is none,
+  // it may be omitted.
   DisplayData display_data = 3;
 }
 
-// A generic node in a bipartite directed hierarchical graph.
-//
-// You can think of this as a GraphNode<NodeT, ConnectionT> where
-// NodeT and ConnectionT are the two types of nodes in the bipartite
-// graph.
-//
-// There is one NodeT on each GraphNode, while the ConnectionT values
-// are named and ordered in the `inputs` and `outputs` fields. For
-// arbitrary graph structures, they are expected to be by-reference.
-//
-// For the Runner API, the type is GraphNode<Transform, PCollection>
-message GraphNode {
+// An applied PTransform! This does not contain the graph data, but only the
+// fields specific to a graph node that is a Runner API transform
+// between PCollections.
+message PTransform {
 
-  // (Required) A URN that describes what kind graph node this is.
+  // (Required) A unique name for the application node.
   //
-  // Specifically, the URN should be enough to decipher the inputs,
-  // outputs, and payload.
-  string urn = 1;
-
-  // (Optional) if this node is contained within a composite, a pointer to the
-  // parent.
-  string parent_id = 2;
+  // Ideally, this should be stable over multiple evolutions of a pipeline
+  // for the purposes of logging and associating pipeline state with a node,
+  // etc.
+  //
+  // If it is not stable, then the runner decides what will happen. But, most
+  // importantly, it must always be here, even if it is autogenerated.
+  string unique_name = 5;
 
-  // (Required) A list of ordered, named inputs to this node.
+  // (Optional) A URN and payload that, together, fully defined the semantics
+  // of this transform.
+  //
+  // If absent, this must be an "anonymous" composite transform.
+  //
+  // For primitive transform in the Runner API, this is required, and the
```

Comment:
I think the set of "known" URNs and their perspective payloads should be enumerated and defined elsewhere, likely in a separate proto file. This can be deferred to a future PR if a TODO is added.